### PR TITLE
fix dupe bug

### PR DESCRIPTION
**Pull request by Raycoms**
Fixes a dupe bug when clicking the button too often.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 1]
```diff
@@ -596,7 +596,7 @@ private void createSkillContent()
         return ImmutableList.copyOf(treeElements);
     }
 
-    private void constructTreeFromRequest(@NotNull final IRequestManager manager, @NotNull final IRequest<?> request, @NotNull final List<RequestWrapper> list, int currentDepth)
+    private void constructTreeFromRequest(@NotNull final IRequestManager manager, @NotNull final IRequest<?> request, @NotNull final List<RequestWrapper> list, final int currentDepth)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java:599]: MAJOR Split this 183 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 56 issues
* ![MAJOR][MAJOR] 53 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L364](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L364): Refactor this method to reduce its Cognitive Complexity from 23 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L373](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L373): Reduce this anonymous class number of lines from 65 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L395](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L395): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L396](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L396): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/f2e0259902ffcee6e74fd39ffb2c28f4e85c69a9/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingTownHall.java [hunk 2]
```diff
@@ -82,7 +82,7 @@ public void onUpgradeComplete(final int newLevel)
      */
     public void addPermissionEvent(final PermissionEvent event)
     {
-        if(getBuildingLevel() >= 1)
+        if(getBuildingLevel() >= 1 && !permissionEvents.contains(event))
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingTownHall.java:85]: MINOR This call to "contains()" may be a performance hot spot if the collection is large. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 58 issues
* ![MAJOR][MAJOR] 53 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L364](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L364): Refactor this method to reduce its Cognitive Complexity from 23 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L373](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L373): Reduce this anonymous class number of lines from 65 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L395](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L395): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L396](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L396): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/41594c428d0c27af0faafad42d5cdcf9d483fa74/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**